tests/unit/frameworks/pytorch/lr_scheduler_test.py

```diff
@@ -53,5 +53,153 @@ def test_chaining_schedulers(self):
+            # 10 batches
+            for _ in range(10):
+                optimizer.step()
+            # Called once per epoch
+            scheduler.step()
+
+        self.assertEqual(expected, actual)
+
+    def test_batch_update(self):
+
+        # OneCycleLR(optimizer, epochs=10, max_lr=2.0, steps_per_epoch=10)
+        # for first 10 epochs
+        expected = [
+            0.08,
+            0.5903,
+            1.5787,
+            1.999,
+            1.8806,
+            1.5878,
+            1.1786,
+            0.734,
+            0.3421,
+            0.0805,
+        ]
+        # StepLR(optimizer, step_size=3, gamma=0.1) for next 10 epochs
+        expected += np.repeat([1.0, 0.1, 0.01, 0.001, 0.0001], 2).tolist()
+
+        # OneCycleLR(optimizer, epochs=10, max_lr=2.0, steps_per_epoch=10)
+        # for last 10 epochs
+        expected += [
+            0.08,
+            0.5903,
+            1.5787,
+            1.999,
+            1.8806,
+            1.5878,
+            1.1786,
+            0.734,
+            0.3421,
+            0.0805,
+        ]
+
+        optimizer = torch.optim.SGD([torch.zeros(1)], lr=1.0)
+        batch_size = 10
+        scheduler = ComposedLRScheduler(
+            optimizer,
+            steps_per_epoch=batch_size,
+            schedulers={
+                0: dict(
+                    lr_scheduler_class=OneCycleLR,
+                    lr_scheduler_args=dict(
+                        epochs=10, max_lr=2.0, steps_per_epoch=batch_size
+                    ),
+                ),
+                10: dict(
+                    lr_scheduler_class=StepLR,
+                    lr_scheduler_args=dict(step_size=2, gamma=0.1),
+                ),
+                20: dict(
+                    lr_scheduler_class=OneCycleLR,
+                    lr_scheduler_args=dict(
+                        epochs=10, max_lr=2.0, steps_per_epoch=batch_size
+                    ),
+                ),
+            },
+        )
+        actual = []
+
+        for _ in range(len(expected)):
+            lr = optimizer.param_groups[0]["lr"]
+            actual.append(round(lr, 4))
+            # 10 batches
+            for _ in range(batch_size):
+                optimizer.step()
+                # Called once per batch
+                scheduler.step()
+
+        self.assertEqual(expected, actual)
+
+    def test_optimizer_update(self):
+        optimizer = torch.optim.SGD(
+            [torch.zeros(1)], lr=1.0, weight_decay=0.0001, momentum=0.0
+        )
+        batch_size = 10
+        expected = [dict(weight_decay=1e-04, momentum=0.0)] * 10
+        expected += [dict(weight_decay=1e-05, momentum=0.9)] * 15
+        scheduler = ComposedLRScheduler(
+            optimizer,
+            steps_per_epoch=batch_size,
+            schedulers={
+                0: dict(
+                    lr_scheduler_class=OneCycleLR,
+                    lr_scheduler_args=dict(
+                        epochs=10,
```

Comment:
It would be better if the number of epochs and the switch point was not the same number as the steps_per_epoch.